File: sanity-flutter/lib/product.dart
class Product {
  final String name;
  final List<dynamic> description;
  final num price;
  final String productimage;

  Product(this.name, this.description, this.price, this.productimage);

  Product.fromJson(Map<String, dynamic> json)
      : name = json['name'],
        description = json['description'],
        price = json['price'],
        productimage = json['productimage'];

  Map<String, dynamic> toJson() => {
        "name": name,
        "description": description,
        "price": price,
        "productimage": productimage
      };
}

File: sanity-ecommerce-poc/schemas/product.js
export default ({
    name: 'product',
    title: "Add Product",
    type: "document",
    fields: [
        {
            title: "Product Name",
            name: "name",
            type: "string",
            validation: Rule => Rule.required().error('Product name is required')
        }, {
            title: "Product Description",
            name: "description",
            type: "array",
            of: [{ type: 'block' }]
        }, {
            title: "Product Price",
            name: "price",
            type: "number",
            validation: Rule => [Rule.required().error('Product price is required'), Rule.precision(2)]
        },
        {
            title: "category",
            name: "category",
            type: "string",
            hidden: true
        },
        {
            title: "Product Image",
            name: "productimage",
            type: "image"
        },
        {
            title: "Product Type",
            name: "type",
            type: "string",
            options: {
                list: [
                    { title: 'camera', value: 'camera' },
                    { title: 'mobile', value: 'mobile' }
                ]
            }
        }
    ],
    initialValue: {
        category: "ecomm-products"
    }
})